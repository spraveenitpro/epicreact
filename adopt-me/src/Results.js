import React from 'react'
import Pet from './Pet'

function Results({ pets }) {
    return (
        <div>
            {!pets.length ? (
                <h1>No Pets found</h1>
            ) : (
                pets.map((pet) => (
                    <Pet
                        name={pet.name}
                        animal={pet.animal}
                        breed={pet.breed}
                        key={pet.id}
                        images={pet.images}
                        location={`${pet.city}, ${pet.state}`}
                        id={pet.id}
                    />
                ))
            )}
        </div>
    )
}

export default Results
