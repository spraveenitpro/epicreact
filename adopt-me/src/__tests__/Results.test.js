/**
 * @jest-environment jsdom
 */

import { expect, test } from '@jest/globals'
import { create } from 'react-test-renderer'
import Results from '../Results'

test('Renders correctly with no pets', () => {
    const tree = create(<Results pets={[]} />).toJSON()
    expect(tree).toMatchSnapshot()
})
