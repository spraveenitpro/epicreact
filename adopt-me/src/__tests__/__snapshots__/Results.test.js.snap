// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly with no pets 1`] = `
<div>
  <h1>
    No Pets found
  </h1>
</div>
`;
